packages/typespec-azure-resource-manager/src/rules/arm-delete-response-codes.ts

```diff
@@ -18,25 +17,27 @@ export const armDeleteResponseCodesRule = createRule({
   },
   create(context) {
     return {
-      model: (model: Model) => {
-        const resources = getArmResources(context.program);
-        const armResource = resources.find((re) => re.typespecType === model);
-        if (armResource && armResource.operations.lifecycle.delete) {
-          const deleteOperation = armResource.operations.lifecycle.delete;
-          const isAsync = getLroMetadata(context.program, deleteOperation.operation) !== undefined;
-          const [httpOp, _] = getHttpOperation(context.program, deleteOperation.operation);
-          const statusCodes = new Set([...httpOp.responses.map((r) => r.statusCodes.toString())]);
-          const expected = new Set(["204", "*"]);
-          expected.add(isAsync ? "202" : "200");
+      root: (program: Program) => {
```

Comment:
This is just a refactoring of this existing rule to be more efficient. It doesn't change the behavior.